src/main/java/org/embulk/input/zendesk/services/ZendeskChatService.java

```diff
@@ -0,0 +1,280 @@
+
+    private ZendeskRestClient zendeskRestClient;
+
+    protected ZendeskInputPlugin.PluginTask task;
+
+    private static final int MAXIMUM_TOTAL_RECORDS = 10000;
+
+    private static final int MAXIMUM_TOTAL_PAGES = 250;
+
+    private static final int MAXIMUM_RECORDS_PER_PAGE = 40;
+
+    public ZendeskChatService(final ZendeskInputPlugin.PluginTask task)
+    {
+        this.task = task;
+    }
+
+    @Override
+    public boolean isSupportIncremental()
+    {
+        return true;
+    }
+
+    @Override
+    public TaskReport addRecordToImporter(final int taskIndex, final RecordImporter recordImporter)
+    {
+        final TaskReport taskReport = Exec.newTaskReport();
+        String startTime = getStartTime();
+        String endTime = getEndTime();
+
+        if (Exec.isPreview()) {
+            fetchData(startTime, endTime, 1, recordImporter);
+            return taskReport;
+        }
+
+        while (true) {
+            String searchURI = buildSearchRequest(startTime, endTime, 1);
+
+            String response = getZendeskRestClient().doGet(searchURI, task, Exec.isPreview());
+
+            JsonNode json = ZendeskUtils.parseJsonObject(response);
+
+            if (!ZendeskUtils.isNull(json)) {
+                int totalRecords = json.get("count").asInt();
+                int totalPages = totalRecords > MAXIMUM_TOTAL_RECORDS
+                    ? MAXIMUM_TOTAL_PAGES
+                    : slicePages(totalRecords);
+
+                logger.info(String.format("Fetching from '%s' to '%s' with '%d' pages and '%d' records",
+                    startTime, endTime, totalPages,  totalRecords));
+
+                if (totalPages > 1) {
+                    final String lastEndTime = endTime;
+                    // Can't start from 0 because page 0 and 1 return the same data, and according to document, page start from 1
+                    IntStream.range(1, totalPages + 1).parallel().forEach(page -> fetchData(startTime, lastEndTime, page, recordImporter));
+                }
+                else {
+                    // in case totalPages 1, IntStream not work
+                    fetchData(startTime, endTime, 1, recordImporter);
+                }
+
+                if (totalRecords <= MAXIMUM_TOTAL_RECORDS || Exec.isPreview()) {
+                    break;
+                }
+
+                endTime = getNextEndDate(startTime, endTime);
+            }
+        }
+
+        storeStartTimeForConfigDiff(taskReport,
+            startTime.equals("0") ? 0 : ZendeskDateUtils.isoToEpochSecond(startTime),
+            ZendeskDateUtils.isoToEpochSecond(endTime));
+        return taskReport;
+    }
+
+    @Override
+    public JsonNode getDataFromPath(final String path, final int page, final boolean isPreview, final long initTime)
+    {
+        String startTime = getStartTime();
+        String endTime = getEndTime();
+
+        List<String> ids = getListIDS(startTime, endTime, 1, true);
```

Comment:
Is the method `getDataFromPath` supposed to be run with preview?

Reply:
No it runs on guess only, preview run here 
https://github.com/treasure-data/embulk-input-zendesk/pull/65/files/7617d292c251a113b781995ea522835a92b26195#diff-6bf3ea9861bae84404dacb2bcbb3e569R61-R63